File: src/azure.py
import os
from openai import AzureOpenAI
from dotenv import load_dotenv

# Load environment variables from .env file
load_dotenv()

# Access environment variables
endpoint = os.getenv("ENDPOINT_URL")
deployment = os.getenv("DEPLOYMENT_NAME")
subscription_key = os.getenv("AZURE_OPENAI_API_KEY")

# Initialize Azure OpenAI client with key-based authentication
client = AzureOpenAI(
    azure_endpoint=endpoint,
    api_key=subscription_key,
    api_version="2024-05-01-preview",
)

# Prepare the initial chat prompt
chat_prompt = [
    {
        "role": "system",
        "content": "You are an AI assistant that helps people find information."
    }
]

# Add a user question to the chat prompt
user_prompt = input("Write your question here: ")
chat_prompt.append({"role": "user", "content": str(user_prompt)})

# Generate the completion
completion = client.chat.completions.create(
    model=deployment,
    messages=chat_prompt,
    max_tokens=800,
    temperature=0.7,
    top_p=0.95,
    frequency_penalty=0,
    presence_penalty=0,
    stop=None,
    stream=False
)

# Extract the response message content using dot notation
reply = completion.choices[0].message.content

# Extract token usage
token_usage = completion.usage

print("ChatGPT's reply:", reply)
print("Token usage:", token_usage)

File: src/Azure.js
import { AzureOpenAI } from "openai";
import dotenv from "dotenv";

dotenv.config();

export async function azure_api(userInput) {
    // Environment variables or fallback values
    const endpoint =
        process.env["AZURE_OPENAI_ENDPOINT"] ||
        "https://projbopenai.openai.azure.com/"; // Update with your endpoint if necessary
    const apiKey =
        process.env["AZURE_OPENAI_API_KEY"] ||
        "<REPLACE_WITH_YOUR_KEY_VALUE_HERE>"; // Update with your actual API key
    const apiVersion = "2024-05-01-preview";
    const deployment = "klemmari-gpt-35-turbo"; // Must match your deployment name

    const client = new AzureOpenAI({
        endpoint,
        apiKey,
        apiVersion,
        deployment,
    });

    // Making a request to the model
    const result = await client.chat.completions.create({
        messages: [
            {
                role: "system",
                content:
                    "You are an AI assistant that helps people find information.",
            },
            {
                role: "user",
                content: userInput, // Use the input passed from the user
            },
        ],
        max_tokens: 800,
        temperature: 0.7,
        top_p: 0.95,
        frequency_penalty: 0,
        presence_penalty: 0,
        stop: null,
    });

    //Log full response from the API
    //console.log("Full response from the API:", result);

    // Loop through the choices and log each one with role and content
    for (const choice of result.choices) {
        console.log("Choice message content:", choice.message.content);
        console.log("Choice message role:", choice.message.role); // This should now give the correct role
    }

    return choice.message.content;
}
